packages/calling/calling/src/CallingClient/CallingClient.test.ts

```diff
@@ -17,26 +16,85 @@ import {
 /* eslint-disable dot-notation */
 import {CallSessionEvent, EVENT_KEYS, MOBIUS_EVENT_KEYS} from '../Events/types';
 import log from '../Logger';
-import {CallingClient, createClient} from './CallingClient';
+import {createClient} from './CallingClient';
 import {ICallingClient} from './types';
 import * as utils from '../common/Utils';
 import {getCallManager} from './calling/callManager';
-import {CALLING_CLIENT_FILE, URL_ENDPOINT} from './constants';
+import {
+  CALLING_CLIENT_FILE,
+  DISCOVERY_URL,
+  NETWORK_CHANGE_DETECTION_UTIL,
+  NETWORK_FLAP_TIMEOUT,
+  REGISTRATION_FILE,
+  SPARK_USER_AGENT,
+  URL_ENDPOINT,
+} from './constants';
 import {MOCK_MULTIPLE_SESSIONS_EVENT, MOCK_SESSION_EVENT} from './callRecordFixtures';
-import * as regUtils from './registration/register';
+import {ILine} from './line/types';
+import {mockPostResponse} from './registration/registerFixtures';
 
 describe('CallingClient Tests', () => {
   // Common initializers
 
-  const handleErrorSpy = jest.spyOn(utils, 'handleRegistrationErrors');
+  const handleErrorSpy = jest.spyOn(utils, 'handleCallingClientErrors');
   const webex = getTestUtilsWebex();
   const defaultServiceIndicator = ServiceIndicator.CALLING;
   const callManager = getCallManager(webex, defaultServiceIndicator);
 
+  const logSpy = jest.spyOn(log, 'info');
+  const warnSpy = jest.spyOn(log, 'warn');
+
+  const mockIPReturnBody = {
+    ipv4: '1.1.1.1',
+    ipv6: '2.2.2.2',
+  };
+
+  const ipPayload = <WebexRequestPayload>(<unknown>{
+    statusCode: 200,
+    body: mockIPReturnBody,
+  });
+
+  const regionBody = {
+    attribution:
+      'This product includes GeoLite2 data created by MaxMind, available from http://www.maxmind.com',
+    clientAddress: '72.163.220.6',
+    clientRegion: 'AP-SOUTHEAST',
+    countryCode: 'IN',
+    disclaimer:
+      'This service is intended for use by Webex Team only. Unauthorized use is prohibited.',
+    regionCode: 'AP-SOUTHEAST',
+    timezone: 'Asia/Kolkata',
+  };
+
+  const regionPayload = <WebexRequestPayload>(<unknown>{
+    statusCode: 200,
+    body: regionBody,
+  });
+
+  const discoveryBody: MobiusServers = getMobiusDiscoveryResponse();
+  const primaryUrl = `${discoveryBody.primary.uris[0]}/calling/web/`;
+  const discoveryPayload = <WebexRequestPayload>(<unknown>{
+    statusCode: 200,
+    body: discoveryBody,
+  });
+
+  const registrationPayload = <WebexRequestPayload>(<unknown>{
+    statusCode: 200,
+    body: mockPostResponse,
+  });
+
+  const uri = `${webex.internal.services._serviceUrls.mobius}${URL_ENDPOINT}`;
+  const myIP = mockIPReturnBody.ipv4;
```

Comment:
We may move these constants to a fixtures file.

Reply:
Done. Created a new `callingClientFixtures.ts` file.